src/js/stores/AppsStore.js

```diff
@@ -80,6 +81,41 @@ function processApps(apps) {
   }).value();
 }
 
+function applyAppDelayStatus(app, queue) {
+  var changed = false;
+  var status;
+
+  var queueEntry = lazy(queue).find(function (entry) {
+    return entry.app != null && app.id === entry.app.id && entry.delay != null;
+  });
+
+  if (queueEntry) {
+    if (queueEntry.delay.overdue === false
+        && queueEntry.delay.timeLeftSeconds > 0) {
+      status = AppStatus.DELAYED;
+    } else if (queueEntry.delay.overdue === true) {
+      status = AppStatus.WAITING;
+    }
+
+    if (status) {
+      if (app.status !== status) {
+        changed = true;
+      }
+      app.status = status;
+    }
+  }
+
+  return changed;
```

Comment:
This should return a boolean always.
I would prefer we do this: `return !!changed;` that way there's no confusion.